CodingConventions.md

```diff
@@ -0,0 +1,200 @@
+# Overview
+
+We would like to be more focused on following the coding conventions popular in .NET community.
+Currently the coding style in the driver code is not very consistent, but the goal is to improve it
+by following coding convention rules.
+
+# Coding conventions
+
+## Writing code
+
+### Static variables
+
+#### Internal or private static member
+
+Use `s_` prefix + camelCase, eg. `s_someVariable`.
+
+```csharp
+public class ExampleClass
+{
+    private static Something s_someVariable;
+    internal static Something s_someVariable2;
+}
+```
+
+#### Internal or private thread static member
+
+Use `t_` prefix + camelCase, eg. `t_someVariable`.
+
+```csharp
+public class ExampleClass
+{
+    [ThreadStatic]
+    private static Something t_someVariable;
+    
+    [ThreadStatic]
+    internal static Something t_someVariable2;
```

Comment:
below such values are named ...Internal 
in this case t_someInternalVariable

Reply:
changed